### PR TITLE
Added Azure DevOps CI/CD definition

### PR DESCRIPTION
**Pull request by OnlyForward**
I configured build pipeline for DeviceAgent.iOS. I added matrix for Mofave-Xcode with properties (IMAGE_POOL and XCODE_Version).

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  - script: make unit-tests
+    displayName: "Make unit-tests"
+
+  - script: |
+      set -e
+      cd cucumber
+      bundle install
+      bundle exec cucumber
+    displayName: "Exec cucumber"
```
Comment (jmoody) [inline azure-pipelines.yml]: In order to get an junit xml report, you need to pass flags to cucumber.

In order for this test to make sense, we need to test against the DeviceAgent we just built.

This step needs to call the bin/ci/cucumber.rb script.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    matrix:
+      Mojave-Xcode-10.2.1:
+        IMAGE_POOL: 'macOS-10.14'
+        XCODE_VERSION: '10.2.1'
+      Mojave-Xcode-11.0:
+        IMAGE_POOL: 'macOS-10.14'
+        XCODE_VERSION: '11'
+  pool:
+    vmImage: $(IMAGE_POOL)
+
```
Comment (jmoody) [inline azure-pipelines.yml]: There are trailing and unnecessary whitespaces in this file.  Can you remove them?

image

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      Mojave-Xcode-11.0:
+        IMAGE_POOL: 'macOS-10.14'
+        XCODE_VERSION: '11'
+  pool:
+    vmImage: $(IMAGE_POOL)
+  
+  steps:
+
+  - script: sudo xcode-select --switch "/Applications/Xcode_$(XCODE_VERSION).app/Contents/Developer"
+    displayName: "Select Xcode $(XCODE_VERSION)"
```
Comment (jmoody) [inline azure-pipelines.yml]: Let's print the Xcode version and some information about the Xcode environment in this step.

```
script: |
  set -e
  sudo xcode-select ...
  echo "Xcode version: $(xcrun xcodebuild -version)"
  echo "Xcode path: $(xcrun -f xcodebuild)"
  echo "Xcode select path: $(xcode-select --print-path)"
```

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  
+  - script: |
+      set -e
+      git clone https://$(GitHubAccessToken)@github.com/xamarinhq/calabash-codesign.git
+      calabash-codesign/apple/create-keychain.sh
+    displayName: "Download and install keychain"
+
```
Comment (jmoody) [inline azure-pipelines.yml]: Let's add a step to make the builds faster.

$ print the ruby version:  `ruby -v`

Install a ~/.gemrc with the following

```
install: --no-document --env-shebang
update:  --no-document --env-shebang
```


and then:

```
bundle install
```

We can rename the step to "Prepare Ruby Environment"

File: azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    displayName: "Make ipa-agent"
+  
+  - script: make test-ipa
+    displayName: "Make test-ipa"
+
+  - script: make test-app
+    displayName: "Make test-app"
+
+  - script: bundle install
+    displayName: "Bundle install"
```
Comment (jmoody) [inline azure-pipelines.yml]: Remove this step.  I should be done in the "Prepare Ruby Environment" step.

File: azure-pipelines.yml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+  - script: |
+      cd cucumber 
+      bundle install
+    displayName: "cd cucumber and bundle install"
```
Comment (jmoody) [inline azure-pipelines.yml]: Remove this step.  This is already done in the bin/ci/cucumber.rb

File: azure-pipelines.yml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+      echo "Xcode path: $(xcrun -f xcodebuild)"
+      echo "Xcode select path: $(xcode-select --print-path)"
+    displayName: "Select Xcode $(XCODE_VERSION)"
+
+
+  - script: |
+      set -e
+      bundle install
+      ruby -v
+      echo "install: --no-document --env-shebang\nupdate:  --no-document --env-shebang" > ~/.gemrc
```
Comment (jmoody) [inline azure-pipelines.yml]: This needs to happen _before_ the `bundle install` call.

This tells `gem` to install gems without documentation which will make installation faster.

**Comment by jmoody** [comment]
@max-lobanov Most of the cucumber tests are failing on Xcode 11.  Is that expected?

File: azure-pipelines.yml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+  - script: make test-app
+    displayName: "Make test-app"
+
+  - script: make unit-tests
+    displayName: "Make unit-tests"
+
+  - script: bundle exec bin/ci/cucumber.rb
+    displayName: "exec cucmber"
+    condition: and(succeeded(), ne(variables['Agent.JobName'], variables['SkipCucumberJob']))
```
Comment (max-lobanov) [inline azure-pipelines.yml]: May be condition `eq(variable['XCODE_VERSION'], '11')` will be clearer?

Comment (ivan-nosar) [reply azure-pipelines.yml]: Done

**Pull request merged**